### PR TITLE
[#IOPID-1838] Edited openapi spec

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-1838] Edited openapi spec](https://github.com/pagopa/io-functions-public/commit/7856afe8fc827f43d7cbf5d16d34be49a6568a83)

#### Motivation and Context

refactor endpoint by extracting the validation part to a RESTy call instead of a redirect-based approach

#### How Has This Been Tested?

not needed. a future PR will address the business logic change

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: openapi/external.yaml [hunk 1]
```diff
@@ -35,7 +36,129 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemJson'
+
+components:
+  schemas:
+    ProfileEmail:
+      type: string
+      format: email
+    GetTokenInfoResponse:
```
Comment (BurnedMarshal) [inline openapi/external.yaml:100]: This Object should transport all the non error scenary like for example `token expired` or `email already validated`. You can use an enum with a `status` prop.

File: openapi/external.yaml [hunk 2]
```diff
@@ -35,7 +36,129 @@ paths:
...
           description: Validation Failed
-        "404":
-          description: Not Found
+  /api/v2/validate-profile-email:
+    get:
+      operationId: getTokenInfo
+      description: |
+        Validates the token and get email relative to a token.
+      parameters:
+        - in: query
```
Comment (BurnedMarshal) [inline openapi/external.yaml:45]: transport this value in header `x-pagopa-email-validation-token`

File: openapi/external.yaml [hunk 3]
```diff
@@ -35,7 +36,129 @@ paths:
...
+      description: |
+        Validates the token and updates the user profile.
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ValidateProfileEmailPayload"
+      responses:
+        "204":
```
Comment (BurnedMarshal) [inline openapi/external.yaml:80]: All the status like for the `get` should be returned in this api

File: openapi/external.yaml [hunk 4]
```diff
@@ -35,7 +36,129 @@ paths:
...
+          description: Email successfully validated.
+        "400":
```
Comment (BurnedMarshal) [inline openapi/external.yaml:82]: add `INVALID_TOKEN` scenario with a 401 status code

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: openapi/external.yaml (hunk 4 above)
Comment (arcogabbo) [reply openapi/external.yaml:92]: done in https://github.com/pagopa/io-functions-public/pull/76/commits/317e91956f0411a4d23dc3131b3cae2bbcb7caff

File: openapi/external.yaml (hunk 3 above)
Comment (arcogabbo) [reply openapi/external.yaml]: changed in https://github.com/pagopa/io-functions-public/pull/76/commits/317e91956f0411a4d23dc3131b3cae2bbcb7caff

File: openapi/external.yaml (hunk 2 above)
Comment (arcogabbo) [reply openapi/external.yaml]: done in https://github.com/pagopa/io-functions-public/pull/76/commits/317e91956f0411a4d23dc3131b3cae2bbcb7caff

File: openapi/external.yaml (hunk 1 above)
Comment (arcogabbo) [reply openapi/external.yaml:113]: changed in https://github.com/pagopa/io-functions-public/pull/76/commits/317e91956f0411a4d23dc3131b3cae2bbcb7caff

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**